freud/diffraction.pyx

```diff
@@ -0,0 +1,348 @@
+import numpy as np
+from scipy import interpolate, ndimage
+import rowan
+
+# from cme_utils.manip import pbc
+# from cme_utils.manip.utilities import rotation_matrix_from_to as cme_rot
+
+
+class Diffraction:
+    def __init__(
+        self,
+        grid_size=512,
+        zoom=4,
+        peak_width=1,
+        length_scale=3.905,
+        bot=4e-6,
+        top=0.7
+    ):
+        """
+        Initialize the diffractometer class
+        Parameters
+        ----------
+        grid_size : int, size of the diffraction grid (default 512)
+        zoom : (default 1)
+        peak_width : (default 1)
+        length_scale : (default 3.905)
+        bot : (default 4e-6)
+        top : (default 0.7)
+        """
+        self.N = grid_size
+        self.zoom = zoom
+        self.peak_width = peak_width
+        self.bin_w = 2.0
+        self.length_scale = length_scale
+        self.bot = bot
+        self.top = top
+
+    def load(self, system):
+        """
+        def load(self, xyz, L):
+            Loads the particle positions and box dimensions for diffraction.
+            Note: only supports orthorhombic boxes
+            Parameters
+            ----------
+            xyz: np.ndarray (N,3), positions of each particle
+            L: iterable object, lengths of box vectors
+            self.box = np.array(
+                    [[L[0], 0.0, 0.0],
+                     [0.0, L[1], 0.0],
+                     [0.0, 0.0, L[2]]]
+                    )
+            self.orig = np.copy(xyz)
+            self.orig, self.image =
+                pbc.shift_pbc(xyz, np.array([L[0], L[1], L[2]]))
+        """
+        box, positions = system
+        self.box = box.to_matrix()
+        self.orig = np.copy(positions)
+
+    def pbc_2d(self, xy, N):
```

Comment:
Any methods that are used internally (but shouldn't be part of the public API) should be made private:
```suggestion
    def _pbc_2d(self, xy, N):
```